pkgs/top-level/perl-packages.nix

```diff
@@ -255,6 +255,7 @@ let
       url = "mirror://cpan/authors/id/P/PL/PLICEASE/Alien-Build-2.37.tar.gz";
       hash = "sha256-MAC8vphIJkP3a7z/zkL9SPJMY6ZFf4qiwWlfSBrJ7VE=";
     };
+    nativeBuildInputs = [ buildPackages.perl ];
```

Comment:
We should provide a comment to this hack and the assignment (which is an implicit override of perl.mini) should only be done for cross-compilation (to document the intention).

Reply:
My thought process behind not guarding was "a full perl is also required for native builds, so this is just more explicit". On the other hand, guarding this with `lib.optional (!stdenv.buildPlatform.canExecute stdenv.hostPlatform)
` also has the additional benefit that it doesn't change the evaluation for native builds, (hopefully) cutting down the number of rebuilds significantly. Will update.

Adding a simple comment like "when cross-compiling, `buildPerlPackage` only appends `perl.mini` to nativeBuildInputs, but this build needs a full perl" to each of these lines should be enough, right?